### PR TITLE
Multiplatform implementation

### PR DESCRIPTION
**Pull request by rmontagnin**
### Changes
- Changed the whole code architecture in order to allow multiplatform usage of the library. 

### Additions
- Added the `ripemd160` module
- Added the `keccak` module
- Added the `extensions` module

### Removals
- I had to remove Gradle Witness as I don't know how to make it work inside multiplatform projects

### CODE REVIEW

**Comment by rmontagnin** [comment]
All the tests should pass correctly and the PR should be ready to be merged

File: README.md [hunk 1]
```diff
@@ -3,65 +3,107 @@
...
+KHash is a [Kotlin multiplatform](https://kotlinlang.org/docs/reference/multiplatform.html) library implementing 
+some of the most common hashing functions used while working with cryptocurrencies of any sort.
 
-Heavily based on this java implementation: [@romus/sha](https://github.com/romus/sha)
+A part from that, it also provides some useful [extensions functions](https://kotlinlang.org/docs/reference/extensions.html).
+
+The implemented hashing functions are the following. 
+* [RIPEMD160](https://en.wikipedia.org/wiki/RIPEMD)
+* [SHA-256](https://en.wikipedia.org/wiki/SHA-2) 
+   - Heavily based on this java implementation: [@romus/sha](https://github.com/romus/sha)
```
Comment (ligi) [inline README.md:15]: following 2 lines belong to SHA-3 not SHA.2

**Review by ligi — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (rmontagnin) [reply README.md:15]: Updated

**Review by ligi — APPROVED** [review]
I see failing tests:


```
khash.sha256.Sha256Tests > testHashRawBytes FAILED
    java.lang.AssertionError at Sha256Tests.kt:36

khash.sha256.Sha256Tests > testDigest FAILED
    java.lang.AssertionError at Sha256Tests.kt:26


```

**Comment by rmontagnin** [comment]
> I see failing tests:
> 
> ```
> khash.sha256.Sha256Tests > testHashRawBytes FAILED
>     java.lang.AssertionError at Sha256Tests.kt:36
> 
> khash.sha256.Sha256Tests > testDigest FAILED
>     java.lang.AssertionError at Sha256Tests.kt:26
> ```

Should be fixed now. I think this was due to a wrong [`mockkObject`](https://mockk.io/#object-mocks) call that made the `Sha256` object return a different result in subsequent tests. With the last commit this should be fixed.

**Comment by ligi** [comment]
dam - just saw now after the merge that the module structure is wrong - sha256 is in sha3 - this is wrong - can you fix @RiccardoM

**Comment by rmontagnin** [comment]
@ligi Implemented the fix inside GH-8

**Pull request merged**